.github/workflows/PullRequestAndMergeMaster.yml

```diff
@@ -0,0 +1,80 @@
+name: PR/Release pipeline
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+  release:
+    types:
+      - created
```

Comment:
I think **published** and **released** work better.  **created** will also trigger just a saving a draft which we might not want that. See here https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#release

For me "published" and "released" worked the best. It only triggers on "releases" and "pre-releases" and when changing form "pre-release" to "release"

Reply:
since the release process also involves using these artifacts in pivnet i will use released and pre-released so we can generate assets before publishing